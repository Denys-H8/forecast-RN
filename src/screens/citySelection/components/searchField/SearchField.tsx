import React from 'react';
import { View, TextInput } from 'react-native';

import Icon from 'react-native-vector-icons/MaterialIcons';
import { styles } from './searchField.styles';

import constants from 'src/constants';

type Props = {
  onChangeText: (val: string) => void;
  value: string;
};

const SearchField: React.FC<Props> = ({ onChangeText, value }) => (
  <View style={styles.container}>
    <View style={styles.inputContainer}>
      <TextInput
        testID="SearchField.TextInput"
        placeholder="Enter the name of city"
        placeholderTextColor={constants.SUB_COLOR}
        style={styles.input}
        onChangeText={(value) => onChangeText(value)}
        value={value}
      />
      <Icon name="search" size={40} color={constants.MAIN_COLOR} />
    </View>
  </View>
);

export default SearchField;
