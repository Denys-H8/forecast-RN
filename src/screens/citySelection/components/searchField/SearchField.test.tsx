import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import SearchField from './SearchField';

const textValue = 'Some text';
const newTextValue = 'New text value';

test('Loading snapshot', () => {
  const onChangeTextMock = jest.fn();

  const { toJSON, getByTestId } = render(<SearchField onChangeText={onChangeTextMock} value={textValue} />);

  fireEvent.changeText(getByTestId('SearchField.TextInput'), newTextValue);

  expect(toJSON()).toMatchSnapshot();
  expect(onChangeTextMock).toHaveBeenCalled();
  expect(onChangeTextMock).toHaveBeenCalledWith(newTextValue);
  expect(onChangeTextMock).not.toHaveBeenCalledWith('Some other value');
});
