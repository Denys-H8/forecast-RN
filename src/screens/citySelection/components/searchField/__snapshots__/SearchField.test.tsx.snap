// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#191C39",
      "paddingBottom": 30,
      "paddingHorizontal": 32,
      "paddingTop": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomWidth": 2,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Enter the name of city"
      placeholderTextColor="#8B8EA5"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#fff",
          "flex": 1,
          "fontFamily": "WorkSans-Regular.ttf",
          "fontSize": 16,
          "marginRight": 10,
        }
      }
      testID="SearchField.TextInput"
      underlineColorAndroid="transparent"
      value="Some text"
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 40,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î¢¶
    </Text>
  </View>
</View>
`;
